Library/Homebrew/diagnostic.rb

```diff
@@ -320,15 +320,8 @@ def check_access_homebrew_repository
       end
 
       def check_access_prefix_directories
-        not_writable_dirs = []
-
-        Keg::ALL_TOP_LEVEL_DIRECTORIES.each do |dir|
-          path = HOMEBREW_PREFIX/dir
-          next unless path.exist?
-          next if path.writable_real?
-          not_writable_dirs << path
-        end
-
+        existing_dirs = Keg::ALL_TOP_AND_SHARE_PATHS.select(&:exist?)
```

Comment:
Every time this is used you check if they exist so maybe it's worth just having the constant having those that exist?